api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/EligibilityAndEntitlementService.java

```diff
@@ -80,4 +87,23 @@ public EligibilityAndEntitlementDecision evaluateClaimantForPaymentCycle(Claiman
                 entitlement, false);
     }
 
+    private CombinedIdentityAndEligibilityResponse getCombinedIdentityAndEligibilityResponse(Claimant claimant,
+                                                                                             EligibilityOutcome eligibilityOverrideOutcome) {
+        if (eligibilityOverrideOutcome == null) {
+            return client.checkIdentityAndEligibility(claimant);
+        } else {
+            return CombinedIdentityAndEligibilityResponse.builder()
+                    .identityStatus(IdentityOutcome.MATCHED)
+                    .eligibilityStatus(eligibilityOverrideOutcome)
+                    .addressLine1Match(VerificationOutcome.NOT_SET)
+                    .deathVerificationFlag(DeathVerificationFlag.N_A)
+                    .dobOfChildrenUnder4(Collections.emptyList())
+                    .emailAddressMatch(VerificationOutcome.NOT_SET)
+                    .mobilePhoneMatch(VerificationOutcome.NOT_SET)
+                    .postcodeMatch(VerificationOutcome.NOT_SET)
+                    .pregnantChildDOBMatch(VerificationOutcome.MATCHED)
+                    .build();
+        }
+    }
```

Comment:
I've had a bit of a tinker with this as it looks a bit clunky, how about this as an alternative?
`    private CombinedIdentityAndEligibilityResponse getCombinedIdentityAndEligibilityResponse(Claimant claimant,
                                                                                             EligibilityOutcome eligibilityOverrideOutcome) {
        if(isOverride(eligibilityOverrideOutcome)) {
            return buildOverridenResponse(eligibilityOverrideOutcome);
        }
        return return client.checkIdentityAndEligibility(claimant);
    }

    private boolean isOverride(EligibilityOutcome eligibilityOverrideOutcome) {
        return eligibilityOverrideOutcome != null;
    }

    private CombinedIdentityAndEligibilityResponse buildOverridenResponse(EligibilityOutcome eligibilityOverrideOutcome) {
        return CombinedIdentityAndEligibilityResponse.builder()
                .identityStatus(IdentityOutcome.MATCHED)
                .eligibilityStatus(eligibilityOverrideOutcome)
                .addressLine1Match(VerificationOutcome.NOT_SET)
                .deathVerificationFlag(DeathVerificationFlag.N_A)
                .dobOfChildrenUnder4(Collections.emptyList())
                .emailAddressMatch(VerificationOutcome.NOT_SET)
                .mobilePhoneMatch(VerificationOutcome.NOT_SET)
                .postcodeMatch(VerificationOutcome.NOT_SET)
                .pregnantChildDOBMatch(VerificationOutcome.MATCHED)
                .build();
    }`

Reply:
applied, thanks